Automattic/wp-calypso
# Add additional WooCommerce onboarding styling to connection flow

This PR implements the remaining missing styles for the WooCommerce/Jetpack onboarding connection flow. This includes the white header bar, width/spacing changes, link and color treatments, and the largest visual change: Muriel/Material styled form components.

It is only enabled in development, is only active through a specific connection flow, and builds upon the previously merged PRs: https://github.com/Automattic/wp-calypso/pull/32993 and https://github.com/Automattic/wp-calypso/pull/33073. 

This PR implements a temporary Muriel styled `TextControl` component in `client/extensions/woocommerce/components` that matches the interface of `@wordpress/components`'s `TextControl` and the `calypso-ui` component proposed at https://github.com/Automattic/wp-calypso/pull/34277. This PR can be switched over to use the `calypso-ui` component, once it is unblocked by https://github.com/WordPress/gutenberg/pull/16227. (The login code is loaded via SSR). This temporary component allows us to test the styles all together, and allows me to keep iterating unblocked on the WooCommerce.com styles next (which are also hosted in Calypso).

Closes https://github.com/woocommerce/woocommerce-admin/issues/2166. Closes https://github.com/woocommerce/woocommerce-admin/issues/2152.

#### Screenshots

<img width="776" alt="Screen Shot 2019-06-28 at 12 03 49 PM" src="https://user-images.githubusercontent.com/689165/60455809-844e3c00-9c05-11e9-9df7-292071ba268e.png">

<img width="776" alt="Screen Shot 2019-06-28 at 12 01 42 PM" src="https://user-images.githubusercontent.com/689165/60455830-8e703a80-9c05-11e9-99df-16624383bf90.png">

#### Testing instructions

* Disconnect Jetpack from your site
* `npm install` to get the updated `color-studio` package
* `npm start`

--

If you have WooCommerce / WooCommerce Admin Installed:

* Enable the onboarding flow in `WooCommerce Admin` and to tell it to point at local Calypso. 
    * Start WooCommerce in development (not plugin) mode
    * Define the following constant in your `wp-config.php` or a plugin file: `define( 'WOOCOMMERCE_CALYPSO_LOCAL', true );`.
    * Make a `POST` request to `/wp-json/wc-admin/v1/onboarding/profile` on your local site, setting `skipped` and `completed` to false.
* Visit `/wp-admin/admin.php?page=wc-admin` on your local site, and the onboarding wizard should be displayed.
* Click `Get Started`, and then `Activate & Continue`.

If you don't want to install WooCommerce Admin:

* Disconnect Jetpack from your site
* Click “Setup Jetpack”.
* On the connection/authorization screen ( https://wordpress.com/jetpack/connect/authorize?client_id= ….. ) copy and paste the whole URL.
* Replace `https://wordpress.com` with ` http://calypso.localhost:3000` and replace the `from` parameter in the URL (it may be something like `connection-banner` or `landing-page-bottom`) with  `from=woocommerce-setup-wizard` . See Pastebin `213d5`  or ping me if you have issues.

---

* Walk through the login and sign-up flows. View both views via the link in the footer. Test typing in the inputs.

* Open `config/development.json` and disable the `jetpack/connect/woocommerce` flag. Refresh the login and sign-up pages and verify the existing Jetpack connection flow appears.
